leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/QueueModeLwM2mRequestSender.java

```diff
@@ -94,8 +94,7 @@ public QueueModeLwM2mRequestSender(PresenceServiceImpl presenceService, LwM2mReq
 
         // If the client is sleeping, warn the user and return
         if (!presenceService.isClientAwake(destination)) {
-            LOG.info("The destination client is sleeping, request couldn't been sent.");
-            return;
+            throw new ClientSleepingException("The destination client is sleeping, request cannot be sent.");
```

Comment:
I'm asking myself if we should return the exception via the ErrorCallback or like you are doing.

My first idea was to use ErrorCallback but maybe you're right and its better like this.

Reply:
FMPOV the `ErrorCallback` throw an exception that indicates the client is sleeping... so `PresenceService` sets the state, and then as soon as a new response tries to be sent, the sleeping exception is thrown immediately. 

It makes sense to me, but it is true that the `ErrorCallback` could also throw the sleeping exception in addition to execute `clientNotResponding` to inform the user directly here and not when a new request tries to be sent. 

So both options are valid for me, either how it is now, or throw the exception both at the begging if the client is sleeping and in the error callback. What do you think is best?

Reply:
let's start like this and we will adjust this later if necessary I will integrate this in queue_mode branch